src/ui/public/visualize/visualize.js

```diff
@@ -144,7 +144,12 @@ uiModules
         }
 
         if (oldVis) $scope.renderbot = null;
-        if (vis) $scope.renderbot = vis.type.createRenderbot(vis, $visEl, $scope.uiState);
+        if (vis) {
+          vis.listeners.renderComplete = function () {
+            $scope.$emit('renderComplete');
+          };
```

Comment:
We shouldn't have an external component be responsible for implementing functionality of the vis-type. We should try and decouple pub/sub.